src/Microsoft.AspNet.OData/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -239,6 +240,27 @@ public static Uri GetNextPageLink(this HttpRequestMessage request, int pageSize,
             return GetNextPageHelper.GetNextPageLink(request.RequestUri, request.GetQueryNameValuePairs(), pageSize, instance, objToSkipTokenValue, options);
         }
 
+        /// <summary>
+        /// Gets the <see cref="ODataOptions"/>.
+        /// </summary>
+        /// <param name="request">The request that provides access to the <see cref="IServiceProvider"/>.</param>
+        /// <returns>The <see cref="ODataOptions"/> instance.</returns>
+        public static ODataOptions GetODataOptions(this HttpRequestMessage request)
+        {
+            return request.GetRequestContainer().GetRequiredService<ODataOptions>();
```

Comment:
>GetRequiredService [](http://example.com/codeflow?start=49&length=18)

Should we make this GetRequiredService, or be prepared to handle nulls?  We add this in ConfigureDefaultServices, which is called from 
MapODataServiceRoute and EnableDependencyInjection, but are there any scenarios where ConfigureDefaultServices might not be called? What about when applying QueryOptions to a non-OData route?

Reply:
Interesting call out re: `QueryOptions` for non-OData routes... I hadn't really considered it. I guess we could make it non-required and return a default instance here if one isn't registered. I'd much rather consumers of this method not have to worry about null checks.